subprojects/core/src/main/java/org/gradle/api/internal/plugins/ImperativeOnlyPluginTarget.java

```diff
@@ -41,6 +49,28 @@ public void applyImperative(@Nullable String pluginId, Plugin<?> plugin) {
         // TODO validate that the plugin accepts this kind of argument
         Plugin<T> cast = uncheckedCast(plugin);
         cast.apply(target);
+
+        // TODO - this should probably be done with TypeMetadataWalker and some sort of annotation handler
+        if (target instanceof ExtensionAware) {
+            ExtensionContainer extensions = ((ExtensionAware) target).getExtensions();
+            Class<?> publicPluginType = GeneratedSubclasses.unpackType(plugin);
+            for (Method method : publicPluginType.getDeclaredMethods()) {
```

Comment:
🤔Question: is it intended that methods declared in the supertypes of the plugin type are ignored by this discovery logic?

Reply:
Sort of.  I think it's not very common for plugins to extend other plugins, so this is probably ok for now.  If we use the TypeMetadata infrastructure, though, which I'd like to try out in the next round of changes, I'm pretty sure it captures methods in supertypes.